test/groupreduce.jl

```diff
@@ -0,0 +1,51 @@
+@kernel function groupreduce_1!(y, x, op, neutral, algo)
+    i = @index(Global)
+    val = i > length(x) ? neutral : x[i]
+    res = @groupreduce(op, val, neutral, algo)
+    i == 1 && (y[1] = res)
+end
+
+@kernel function groupreduce_2!(y, x, op, neutral, algo, ::Val{groupsize}) where {groupsize}
+    i = @index(Global)
+    val = i > length(x) ? neutral : x[i]
+    res = @groupreduce(op, val, neutral, algo, groupsize)
+    i == 1 && (y[1] = res)
+end
```

Comment:
These need to be `cpu=false` since you are using non-top-level `@synchronize`

Reply:
Done.